src/deploy.js

```diff
@@ -1,15 +1,85 @@
-import { spawn } from './utils';
+import AWS from 'aws-sdk';
+import s3 from 's3-node-client';
+import dotenv from 'dotenv';
+
+const path = require('path');
+
+// default build directory
+const BUILD = 'build/';
+
+/**
+ * Returns an object with all variables loaded from a environment
+ * @param {string} environmentName is the suffix after .env.*
+ */
 
 const deploy = async (opts) => {
-  const { path } = opts;
-
-  console.error(`path passed '${path}'`);
-  console.log(process.cwd());
-  try {
-    await spawn('./src/test.sh');
-  } catch (e) {
-    console.error(e);
-  }
+  // const { path } = opts;
+
+  const usageMessage = console.log(
+    'usage: $0 [-e <path/to/file>] [-v <version string>] [-b <path/to/build>]',
+  );
+
+  console.log(`Exectued with path: ${opts.path}`);
+  console.log(usageMessage);
+
+  // fetch environment variables
+  dotenv.config({ path: path.resolve(process.cwd(), '.env.dev') });
+  /* eslint-disable no-unused-vars */
+  const {
+    REACT_APP_GRAASP_DEVELOPER_ID,
+    REACT_APP_GRAASP_APP_ID,
+    REACT_APP_GRAASP_DOMAIN,
+    REACT_APP_HOST,
+    REACT_APP_VERSION,
+    REACT_APP_BASE,
+    NODE_ENV,
+    BUCKET,
+    AWS_DEFAULT_REGION,
+    AWS_ACCESS_KEY_ID,
+    AWS_SECRET_ACCESS_KEY,
+    DISTRIBUTION,
+  } = process.env;
+  /* eslint-enable no-unused-vars */
+
+  AWS.config.getCredentials(function (err) {
```

Comment:
Are you using this? Maybe promisify in order not to have to shove everything inside the callback.

Reply:
Yes. This reloads the credentials from the `.env.xxx` set before. Hence, the below dev note in the [sdk doc](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/Config.html#getCredentials-property) does not apply.

> Note: If you configure the SDK with static or environment credentials, the credential data should already be present in credentials attribute. This method is primarily necessary to load credentials from asynchronous sources, or sources that can refresh credentials periodically.

However, I'm still looking for a proper way to refactor this parts and maybe `promisify` will help me to do so.

Reply:
By rethinking the look we already assure that the configuration can be loaded since we're manually checking that the env variables (`AWS_ACCESS_KEY_ID,  AWS_SECRET_ACCESS_KEY`) exist. I will implement a first draft that we might discuss during the next meeting.